devito/symbolics/inspection.py

```diff
@@ -63,6 +63,22 @@ def count(exprs, query):
     return dict(mapper)
 
 
+def retrieve_dim(dim):
```

Comment:
Oh sorry typed to quick, this should be in `search.py` as 
```
def retreive_dimension(exprs, mode='all', deep=False):
    """Shorthand to retrieve the dimensions in ``exprs``."""
    return search(exprs, q_dimension mode, 'dfs', deep)
```
 (rename `retreive_dimension`)  and then need the `q_dimension` in `queries.py` 

```
def q_dimension(expr):
    from devito.types.dimension import Dimension
    return isinstance(expr, Dimension)
```